job-server/src/spark.jobserver/WebApi.scala

```diff
@@ -273,6 +273,33 @@ class WebApi(system: ActorSystem,
               }
             }
           }
+        } ~
+        put {
+          parameter("reset") { reset =>
+            respondWithMediaType(MediaTypes.`text/plain`) { ctx =>
+              reset match {
+                case "all" => {
+                  import ContextSupervisor._
+                  import collection.JavaConverters._
+                  logger.warn("refreshing contexts")
+                  val future = (supervisor ? ListContexts).mapTo[Seq[String]]
+                  Await.result(future, 1 second).asInstanceOf[Seq[String]]
+                    .foreach { context => {
+                      val stopFuture = supervisor ? StopContext(context)
+                      Await.ready(stopFuture, 5 seconds)
+                    }
+                  }
+                  Thread.sleep(1000) // we apparently need some sleeping in here, so spark can catch up
+
+                  (supervisor ? AddContextsFromConfig).onFailure {
```

Comment:
Good point, the current behavior is more of a "restart", restoring the contexts to what they would look like if you re-started the job server.